gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java

```diff
@@ -337,7 +356,8 @@ private void initialize(Dag<JobExecutionPlan> dag)
       //Are there any jobs already in the running state? This check is for Dags already running
       //before a leadership change occurs.
       for (DagNode<JobExecutionPlan> dagNode : dag.getNodes()) {
-        if (DagManagerUtils.getExecutionStatus(dagNode) == RUNNING) {
+        if (DagManagerUtils.getExecutionStatus(dagNode) == RUNNING ||
+            DagManagerUtils.getExecutionStatus(dagNode) == PENDING) {
```

Comment:
see comment above. Unless I am missing something, this may result in a job not getting submitted.

Reply:
I think the PENDING check is unnecessary. The DagNode will have execution status ExecutionStatus.UNKNOWN.